frontend/src/main/components/PersonalSchedules/PersonalSchedulePanel.js

```diff
@@ -0,0 +1,189 @@
+export default function SchedulerPanel({
+  Events = [],
+  eventColor = "#d1ecf188",
+  borderColor = "#bee5eb",
+}) {
+  // Stryker restore all
+  const testId = "SchedulerPanel";
+
+  return (
+    <Container fluid style={styles.schedulerPanel}>
+      <Row style={styles.headerRow}>
+        <Col style={styles.timeColumn}></Col>
+        {daysOfWeek.map((day) => (
+          <Col
+            key={day}
+            style={styles.dayColumn}
+            data-testid={`${testId}-${day}-column`}
+          >
+            <Card style={styles.dayCard}>
+              <Card.Body>
+                <Card.Title
+                  style={styles.dayTitle}
+                  data-testid={`${testId}-${day}-title`}
+                >
+                  {day}
+                </Card.Title>
+              </Card.Body>
+              {Events.filter((event) => event.day === day).map((event) => (
+                <SchedulerEvents
+                  key={event.id}
+                  event={event}
+                  eventColor={eventColor}
+                  borderColor={borderColor}
+                />
+              ))}
+            </Card>
+          </Col>
+        ))}
+      </Row>
+      <Row>
+        <Col style={styles.timeColumn}>
+          {
+            <div
+              /* Stryker disable next-line all : no test needed for styling */
+              style={{ ...styles.timeSlot, height: "30px", border: "0" }}
+            ></div>
+          }
+          {hours.map((hour, index) => (
+            <div
+              key={index}
+              /* Stryker disable next-line all : no test needed for styling */
+              style={{ ...styles.timeSlot, border: "0" }}
+              data-testid={`${testId}-${hour.replace(" ", "-")}-title`}
+            >
+              <span
+                style={styles.hourLabel}
+                data-testid={`${testId}-${hour.replace(" ", "-")}-label`}
+              >
+                {hour}
+              </span>
+            </div>
+          ))}
+        </Col>
+
+        {
+          // Stryker disable all : no test needed for styling
+          daysOfWeek.map((day) => (
+            <Col key={day} style={styles.dayColumn}>
+              {<div style={{ ...styles.timeSlot, height: "30px" }}></div>}
+              {hours.slice(0, hours.length - 1).map((hour) => (
+                <div
+                  key={hour}
+                  style={styles.timeSlot}
+                  data-testid={`${testId}-base-slot`}
+                >
+                  <Card style={styles.eventCard} />
+                </div>
+                // Stryker restore all
+              ))}
+            </Col>
+          ))
+        }
+      </Row>
+    </Container>
+  );
+}
+
+// Stryker disable all: no need to test styles
+const styles = {
+  schedulerPanel: {
+    backgroundColor: "#fff",
+    padding: "20px",
+  },
+  headerRow: {
+    textAlign: "center",
+  },
+  timeColumn: {
+    textAlign: "right",
+    borderRight: "1px solid #ddd",
```

Comment:
Can we put all of this into a CSS file instead of here in the React code?

Reply:
I still want to see these factored out into index.css unless that simply cannot possible be done (e.g. the aspects of styles that are computed programmatically such an exact x or y position).